### PR TITLE
feat: create new card from smart picker

### PR DESCRIPTION
**Pull request by luka-nextcloud**
* Resolves: #4962
* Target version: main

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] Sign-off message is added to all commits
- [ ] Tests (unit, integration, api and/or acceptance) are included
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by juliushaertl** [comment]
As discussed during the call:


  
- [ ] Put card title to the top
- [ ] Board picker and list picker side by side
- [ ] Make `CardSidebarTabDetails` sections reusable 
  - [ ] Migrate `CreateNewCardCustomPicker` to use a boilerplate card object as well instead of individual duedate/selectedLabels/.. data attributes
  - [ ] Extract tag multiselect section with icon from `CardSidebarTabDetails` to a sepeate component `TagSelector`
  		- those get a card object passed `<DueDateSelector :card="card" />`
  	- [ ] Use the `TagSelector` in `CardSidebarTabDetails`
  	- [ ] Use the `TagSelector` in `CreateNewCardCustomPicker`
  - [ ] Do the same for `AssignmentSelector`, `DueDateSelector`
  - [ ] Use Description component in `CreateNewCardCustomPicker`
  
As the last step:		
		
- [ ] Merge CreateCardDialog and CreateNewCardCustomPicker (only use the new one with all options)
  - Old create card dialog can just have a prop to show the "Card was added to the board" info or emit the event for the smart picker

File: src/views/CreateNewCardCustomPicker.vue [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+					:can-edit="!loading && selectedBoard"
+					@select="onSelectUser"
+					@remove="onRemoveUser" />
+			</div>
+
+			<div class="section-details">
+				<DueDateSelector :card="card" :can-edit="canEdit" @change="updateCardDue" />
+			</div>
+
+			<textarea v-model="pendingDescription" :disabled="loading || !selectedStack" :placeholder="t('deck', 'Card description')" />
```
Comment (juliushaertl) [inline src/views/CreateNewCardCustomPicker.vue:65]: This should switch to use the `Description` component so we also have the rich text editor

File: src/views/CreateNewCardCustomPicker.vue [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+						:placeholder="t('deck', 'Select a list')"
+						:options="stacksFromBoard"
+						:max-height="100"
+						:disabled="loading || !selectedBoard"
+						class="multiselect-list"
+						label="title" />
+				</div>
+			</div>
+
+			<TagSelector :card="card"
```
[the remark this answers is not in the document]
Comment (juliushaertl) [reply src/views/CreateNewCardCustomPicker.vue]: > maybe we can add a CardDetailRow component which has a slot for the selector and a slot for the icon to render them like this:

I think we can even just add the icons directly to the individual components, probably no need for a separate wrapper then

File: src/views/CreateNewCardCustomPicker.vue [hunk 3]
```diff
@@ -0,0 +1,340 @@
+<template>
+	<div class="modal-scroller">
+		<div v-if="!creating && !created" id="modal-inner" :class="{ 'icon-loading': loading }">
+			<h3>{{ t('deck', 'Create a new card') }}</h3>
+
+			<input v-model="card.title"
```
Comment (juliushaertl) [inline src/views/CreateNewCardCustomPicker.vue:6]: This input should probably have an autofocus :)

File: src/init-reference.js [hunk 4]
```diff
@@ -82,3 +82,19 @@ registerWidget('deck-comment', (el, { richObjectType, richObject, accessible })
 		},
 	}).$mount(el)
 })
+
+registerCustomPickerElement('create-new-deck-card', async (el, { providerId, accessible }) => {
```
Comment (juliushaertl) [inline src/init-reference.js:86]: Let's also pass `normal` for the size here to match the requirements from https://github.com/nextcloud/deck/issues/4962#issuecomment-1661721341 

Reference: https://github.com/nextcloud-libraries/nextcloud-vue/blob/8dbc4451dfef7b0edf7ed438f26b2064ffddd113/src/components/NcRichText/NcReferencePicker/customPickerElements.js#L28

**Comment by luka-nextcloud** [comment]
@juliushaertl I've done the following things:

- add icons before selectors input
- add prop autofocus for title input
- pass size param 'normal' while calling registerCustomPickerElement()
- fix closing create card modal

**Review by juliushaertl — APPROVED** [review]

**Comment by juliushaertl** [comment]
@luka-nextcloud FYI I pushed a bit more polishing, to reuse styles and also use the same component for creating files from the talk app 

Latest state:

Screenshot 2023-08-09 at 13 35 00

**Pull request merged**